extension/src/experiments/webview/messages.ts

```diff
@@ -74,8 +75,11 @@ export class WebviewMessages {
     this.addStage = addStage
   }
 
-  public async changeHasConfig() {
-    this.hasConfig = !!(await this.hasStages())
+  public async changeHasConfig(update?: boolean) {
+    const stages = await this.hasStages()
+    this.hasValidDvcYaml = stages !== undefined
+    this.hasConfig = !!stages
```

Comment:
[Q] Is there any way that `this.hasValidDvcYaml` isn't `!this.hasStages`? Is `this.hasValidDvcYaml === !this.hasConfig` always true? What values of stages mean that we need both variables? Can we refactor to make it more obvious? (my brain is tiny)

Reply:
`dvc stage list` will return '' if no pipelines were found. It will return an error (`'./dvc.yaml' validation failed.`) if the `dvc.yaml` file is broken. the `listStages` command will return the result, but it has a `catch` clause that will return undefined. 

In other words, `hasInvalidDvcYaml = hasStages() === undefined` and `doesNotHaveConfig = hasStages() === ''`. 

We cannot test accurately for `hasConfig` if `hasValidDvcYaml` is `true`, but I chose to set it to false because it shows the message at the bottom of the table.